asllib/doc/ASLSemanticsReference.tex

```diff
@@ -1056,6010 +947,6041 @@ \chapter{Reading guide}
+    \item evaluating the local declaration $\ldi$ with $\langle\vm\rangle$ as the initializing
+    value in $\envone$ as per Chapter~\ref{chap:eval_local_decl} is $\Normal(\newg, \newenv)$;
+    \item the result of the entire evaluation is $\Continuing(\newg, \newenv)$.
     \end{itemize}
 
-    \subsection{Example: SemanticsRule.LDDiscard.asl}
-    In the specification:
-    \VerbatimInput{../tests/ASLSemanticsReference.t/SemanticsRule.LDDiscard.asl}
-    \texttt{var - : integer;} does not modify the environment.
+    \subsection{Example: SemanticsRule.SDeclSome.asl}
+    The specification:
+    \VerbatimInput{../tests/ASLSemanticsReference.t/SemanticsRule.SDeclSome.asl}
+    \texttt{let x = 3;} binds \texttt{x} to \texttt{3} in the empty environment.
 
-  \subsection{Code}
-  \VerbatimInput[firstline=\LDDiscardBegin, lastline=\LDDiscardEnd]{../Interpreter.ml}
+
+  \CodeSubsection{\SDeclSomeBegin}{\SDeclSomeEnd}{../Interpreter.ml}
 
 \begin{emptyformal}
   \subsection{Formally}
 \begin{mathpar}
-  \inferrule{}
+  \inferrule{
+    \evalexpr{\env, \ve} \evalarrow \Normal(\vm, \envone) \OrAbnormal\\
+    \evallocaldecl{\envone, \ldi, \langle\vm\rangle} \evalarrow \Normal(\newg, \newenv)\\
+  }
   {
-    \evallocaldecl{\env, \texttt{LDI\_Discard}, \Ignore} \evalarrow \Normal(\emptygraph, \env)
+    \evalstmt{\env, \texttt{S\_Decl}(\Ignore, \ldi, \langle\ve\rangle)} \evalarrow \Continuing(\newg, \newenv)
   }
 \end{mathpar}
 \end{emptyformal}
 
 \isempty{\subsection{Comments}}
 
-\section{SemanticsRule.LDVar \label{sec:SemanticsRule.LDVar}}
+\section{SemanticsRule.SDeclNone \label{sec:SemanticsRule.SDeclNone}}
     \subsection{Prose}
-    Evaluation of the local variables $\ldi$ in the environment \\
-    $\env$ is $\Normal(\vg, \newenv)$ and all of the following apply:
+    Evaluation of the statement $\vs$ in an environment $\env$ is
+    either $\Continuing(\newg, \newenv)$ or an abnoraml configuration,
+    and all of the following apply:
     \begin{itemize}
-    \item $\ldi$ is a variable declaration, $\texttt{LDI\_Var}(\vx)$;
-    \item $\minitopt$ is $\vm$;
-    \item $\vm$ is a pair consisting of the value $\vv$ and execution graph $\vgone$;
-    \item declaring $\vx$ in $\env$ is $(\newenv, \vgtwo)$;
-    \item $\newg$ is the ordered composition of $\vgone$ and $\vgtwo$ with the $\asldata$ edge.
+    \item $\vs$ is a declaration without an initial value, $\texttt{S\_Decl}(\Ignore, \ldi, \None)$;
+    \item evaluating the local declaration $(\ldi, \None)$ as per Chapter~\ref{chap:eval_local_decl}
+    is \\ $\Normal(\newg, \newenv)$;
+    \item the result of the entire evaluation is $\Continuing(\newg, \newenv)$.
     \end{itemize}
 
-    \subsection{Example: SemanticsRule.LDVar0.asl}
+    \subsection{Example: SemanticsRule.SDeclNone.asl}
     In the specification:
-    \VerbatimInput{../tests/ASLSemanticsReference.t/SemanticsRule.LDVar0.asl}
-    \texttt{var x = 3;} binds \texttt{x} to the evaluation of \texttt{3} in $\env
```

Comment:
if we do need both the One of the following and the All of the following, please might you indent the second one?
Also All of the following applies -> all of the following apply